### PR TITLE
cloud/dynamiccontroller: fix toBytes

### PR DESCRIPTION
**Pull request by Iceber**
Signed-off-by: Iceber Gu <wei.cai-nat@daocloud.io>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
`klog.Fatal`cannot be called even if marshal content to []byte failed.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cloud/pkg/dynamiccontroller/application/application.go [hunk 1]
```diff
@@ -341,22 +341,20 @@ func NewApplicationCenter(dynamicSharedInformerFactory dynamicinformer.DynamicSh
...
-	var bytes []byte
+
 	var err error
-	switch i := i.(type) {
-	case []byte:
-		bytes = i
-	default:
-		bytes, err = json.Marshal(i)
-		if err != nil {
-			klog.Fatalf("marshal content to []byte failed, err: %v", err)
```
Comment (Iceber) [inline cloud/pkg/dynamiccontroller/application/application.go:356]: klog.Fatalcannot be called even if marshal content to []byte failed.

**Review by GsssC — APPROVED** [review]

**Comment by GsssC** [comment]
Thanks for your contribution! : )

File: cloud/pkg/dynamiccontroller/application/application.go [hunk 2]
```diff
@@ -341,22 +341,20 @@ func NewApplicationCenter(dynamicSharedInformerFactory dynamicinformer.DynamicSh
...
-		}
+	if bytes, err = json.Marshal(i); err != nil {
+		klog.Errorf("marshal content to []byte failed, err: %v", err)
 	}
-	return bytes
+	return
```
Comment (fisherxu) [inline cloud/pkg/dynamiccontroller/application/application.go:357]: lgtm, the return here maybe no needed?

Comment (Iceber) [reply cloud/pkg/dynamiccontroller/application/application.go:357]: In golang, if there is a return value, then `return` is required.

Comment (fisherxu) [reply cloud/pkg/dynamiccontroller/application/application.go:357]: Okey, get it.
/approve

**Pull request merged**